libs/platform/kernel/ebpf_platform_kernel.c

```diff
@@ -299,4 +302,60 @@ ebpf_log_function(void* context, const char* format_string, ...)
 
     va_end(arg_start);
     return 0;
+}
+
+ebpf_error_code_t
+ebpf_access_check(
+    ebpf_security_descriptor_t* security_descriptor,
+    ebpf_security_access_mask_t request_access,
+    ebpf_security_generic_mapping_t* generic_mapping)
+{
+    ebpf_error_code_t result;
+    NTSTATUS status;
+    SECURITY_SUBJECT_CONTEXT subject_context = {0};
+    DWORD granted_access;
+
+    SeCaptureSubjectContext(&subject_context);
+    SeLockSubjectContext(&subject_context);
+    if (!SeAccessCheck(
+            security_descriptor,
+            &subject_context,
+            true,
+            request_access,
+            0,
+            NULL,
+            generic_mapping,
+            KernelMode,
+            &granted_access,
+            &status))
+        result = EBPF_ERROR_ACCESS_DENIED;
+    else {
```

Comment:
seems inconsistent to use {} around one line below but not around one line above.  Recommend adding it above for consistency.